README.md
# NAME

Pod::Weaver::PluginBundle::Author::CSSON - Weave Pod like CSSON

# VERSION

Version 0.1101, released 2016-02-18.

# STATUS

Deprecated. See [Dist::Iller::Config::Author::CSSON](https://metacpan.org/pod/Dist::Iller::Config::Author::CSSON) instead.

# SOURCE

[https://github.com/Csson/p5-Pod-Weaver-PluginBundle-Author-CSSON](https://github.com/Csson/p5-Pod-Weaver-PluginBundle-Author-CSSON)

# HOMEPAGE

[https://metacpan.org/release/Pod-Weaver-PluginBundle-Author-CSSON](https://metacpan.org/release/Pod-Weaver-PluginBundle-Author-CSSON)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Pod-Weaver-PluginBundle-Author-CSSON

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.1101 2016-02-18T23:18:49Z UTC
 [Other Changes]
 - Deprecated. Will be removed shortly.

 [Stats & Dependency Changes]
 - 5 files changed, 47 insertions(+), 22 deletions(-)
 - (run req) + Pod::Weaver::Config::Assembler (any)
 - (run req) ~ Path::Tiny (any) --> 0.072
 - (run req) ~ perl 5.010001 --> 5.010000
 - (run req) - Pod::Weaver
 - (test req) + ExtUtils::MakeMaker (any)
 - (test req) + File::Spec (any)
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (test req) - Test::NoTabs
 - (test rec) + CPAN::Meta 2.120900
 - (conf req) + ExtUtils::MakeMaker (any)
 - (conf req) - Module::Build::Tiny
 - (dev req) + Dist::Zilla 5.000
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
 - (dev req) + Dist::Zilla::Plugin::CheckChangesHasContent 0.008
 - (dev req) + Dist::Zilla::Plugin::Clean 0.07
 - (dev req) + Dist::Zilla::Plugin::ConfirmRelease (any)
 - (dev req) + Dist::Zilla::Plugin::CopyFilesFromBuild 0.151680
 - (dev req) + Dist::Zilla::Plugin::Deprecated (any)
 - (dev req) + Dist::Zilla::Plugin::DistIller::MetaGeneratedBy (any)
 - (dev req) + Dist::Zilla::Plugin::ExecDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git 2.036
 - (dev req) + Dist::Zilla::Plugin::Git::Check (any)
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Contributors 0.017
 - (dev req) + Dist::Zilla::Plugin::Git::GatherDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Push (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Tag (any)
 - (dev req) + Dist::Zilla::Plugin::GithubMeta 0.54
 - (dev req) + Dist::Zilla::Plugin::InstallRelease 0.008
 - (dev req) + Dist::Zilla::Plugin::License (any)
 - (dev req) + Dist::Zilla::Plugin::MakeMaker (any)
 - (dev req) + Dist::Zilla::Plugin::Manifest (any)
 - (dev req) + Dist::Zilla::Plugin::ManifestSkip (any)
 - (dev req) + Dist::Zilla::Plugin::MetaJSON (any)
 - (dev req) + Dist::Zilla::Plugin::MetaNoIndex (any)
 - (dev req) + Dist::Zilla::Plugin::MetaProvides::Package 2.003001
 - (dev req) + Dist::Zilla::Plugin::MetaYAML (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::PodSyntaxTests (any)
 - (dev req) + Dist::Zilla::Plugin::PodWeaver 4.006
 - (dev req) + Dist::Zilla::Plugin::Prereqs (any)
 - (dev req) + Dist::Zilla::Plugin::Readme (any)
 - (dev req) + Dist::Zilla::Plugin::ReadmeAnyFromPod 0.150250
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::RunExtraTests 0.028
 - (dev req) + Dist::Zilla::Plugin::ShareDir (any)
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.054
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::Kwalitee::Extra v0.2.1
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::ReportPrereqs 0.021
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.05
 - (dev req) + Dist::Zilla::Plugin::TestRelease (any)
 - (dev req) + Dist::Zilla::Plugin::TravisYML 1.15
 - (dev req) + Dist::Zilla::Plugin::UploadToCPAN (any)
 - (dev req) + Pod::Elemental::Transformer::List (any)
 - (dev req) + Pod::Weaver::Plugin::SingleEncoding (any)
 - (dev req) + Pod::Weaver::Plugin::Transformer (any)
 - (dev req) + Pod::Weaver::PluginBundle::CorePrep (any)
 - (dev req) + Pod::Weaver::Section::Authors (any)
 - (dev req) + Pod::Weaver::Section::Collect (any)
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Pod::Weaver::Section::Generic (any)
 - (dev req) + Pod::Weaver::Section::Leftovers (any)
 - (dev req) + Pod::Weaver::Section::Legal (any)
 - (dev req) + Pod::Weaver::Section::Name (any)
 - (dev req) + Pod::Weaver::Section::Region (any)
 - (dev req) + Pod::Weaver::Section::Version (any)
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Pod 1.41
 - (dev req) + Test::Version 1
 - (dev req) + Test::Warnings 0.026
 - (dev sug) + Dist::Iller 0.1407
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0309

0.1100 2015-01-07T21:55:10Z CET
 [Enhancements]
 - Expand Dist Zilla bundle before release.

0.1000 2015-01-07T00:11:01Z CET
 - Initial release


lib/Pod/Weaver/PluginBundle/Author/CSSON.pm
use 5.10.1;
use strict;
use warnings;

package Pod::Weaver::PluginBundle::Author::CSSON;

# ABSTRACT: Weave Pod like CSSON
# AUTHORITY
our $VERSION = '0.1102';

use strict;
use warnings;
use Pod::Weaver::Config::Assembler;
use Path::Tiny;

sub xp {
    Pod::Weaver::Config::Assembler->expand_package(shift);
}

sub mvp_bundle_config {
    my @plugins = ();

    # check git config
    my $include_default_github = 0;
    my $git_config = path('.git/config');
    if($git_config->exists) {
        my $git_config_contents = $git_config->slurp_utf8;
        if($git_config_contents =~ m{github\.com:([^/]+)/(.+)\.git}) {
            $include_default_github = 1;
        }
        else {
            warn ('[PW/@Author: ] No github url found');
        }
    }

    push @plugins => (
        ['@Author::CSSON/CorePrep',       xp('@CorePrep'),       { } ],
        ['@Author::CSSON/SingleEncoding', xp('-SingleEncoding'), { } ],
        ['@Author::CSSON/Name',           xp('Name'),            { } ],
        ['@Author::CSSON/Version',        xp('Version'),         { format => q{Version %v, released %{YYYY-MM-dd}d.} } ],
        ['@Author::CSSON/Prelude',        xp('Region'),          { region_name => 'prelude' } ],
    );

    foreach my $plugin (qw/Synopsis Description Overview Stability/) {
        push @plugins => ['@Author::CSSON/'.$plugin, xp('Generic'), { header => uc $plugin } ];
    }

    foreach my $plugin ( ['Attributes', 'attr'],
                         ['Methods', 'method'],
                         ['Functions', 'func'],
    ) {
        push @plugins => [ $plugin->[0], xp('Collect'), { command => $plugin->[1], header => uc $plugin->[0] } ];
    }
    push @plugins => (
        ['@Author::CSSON/Leftovers',             xp('Leftovers'), { } ],
        ['@Author::CSSON/postlude',              xp('Region'),    { } ],
        (
            !$ENV{'ILLER_MINTING'} && $include_default_github ?
            ['@Author::CSSON/Source::DefaultGitHub', xp('Source::DefaultGitHub'), { text => 'L<%s>' } ]
            :
            ()
        ),
        ['@Author::CSSON/Homepage::DefaultCPAN', xp('Homepage::DefaultCPAN'), { text => 'L<%s>' } ],
        ['@Author::CSSON/Authors',               xp('Authors'),   { } ],
        ['@Author::CSSON/Legal',                 xp('Legal'),     { } ],

        ['@Author::CSSON/List', xp('-Transformer'), { transformer => 'List' } ],
        ['@Author::CSSON/Splint', xp('-Transformer'), { transformer => 'Splint' } ],
    );

    return @plugins;
}

1;

__END__

=head1 STATUS

Deprecated. See L<Dist::Iller::Config::Author::CSSON> instead.

=cut


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.043.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Weave Pod like CSSON",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Pod-Weaver-PluginBundle-Author-CSSON",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Pod::Weaver::PluginBundle::Author::CSSON",
  "PREREQ_PM" => {
    "Path::Tiny" => "0.072",
    "Pod::Weaver::Config::Assembler" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.1102",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Path::Tiny" => "0.072",
  "Pod::Weaver::Config::Assembler" => 0,
  "Test::More" => "0.96"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
